lms/services/roster.py

```diff
@@ -111,11 +112,33 @@ def fetch_assignment_roster(self, assignment: Assignment) -> None:
         ), "Trying fetch roster for course without service URL."
         lti_registration = self._get_lti_registration(lms_course)
 
-        roster = self._lti_names_roles_service.get_context_memberships(
-            lti_registration,
-            lms_course.lti_context_memberships_url,
-            resource_link_id=assignment.lti_v13_resource_link_id,
-        )
+        try:
+            roster = self._lti_names_roles_service.get_context_memberships(
+                lti_registration,
+                lms_course.lti_context_memberships_url,
+                resource_link_id=assignment.lti_v13_resource_link_id,
+            )
+        except ExternalRequestError as err:
+            if (
+                err.response_body
```

Comment:
This is what pylint is missing (see changes below in pyproject.toml), 

`err.response_body` is `str | None` but this condition narrows the type to str which pylint doesn't detect.